visualize-admin/visualization-tool
# feat: Add layout step

This PR introduces the layouting step to the application when user created more than one chart at once in the editor mode (hidden behind `flag__layoutStep=true`). Also see [this Notion document](https://www.notion.so/interactivethings/Compositions-218e4f5e1b514a3e8930a7a8cab2092a).

- Added layouting step, conditionally enabled when there are at least two charts
- Separated panel layout for every step, as they do not share the same "left-middle" layout anymore and made it more configurable (`L`, `M`, `R`)
- Introduced `LayoutConfigurator`
  - Based on `ConfiguratorState`'s `layout` property
  - Made it possible to set layout annotations (title and description) by moving the meta property from `ConfiguratorState` to `layout`
  - Made it possible to select either Tab or Dashboard layout (with Vertical and Tall sub layouts for the Dashboard layout)
- Adapted paddings and general styles to match the design

### Next steps
- [ ] Tweak annotations fonts and background color in published charts
- [ ] Add translations for "Vertical", "Tall"
- [ ] To be confirmed: use `PanelHeader` in `CONFIGURING_CHART` step instead of keeping top buttons in drawer & chart tabs
- [ ] Check how to handle responsiveness of tall layout with @KerstinFaye
- [ ] Make axis titles responsive
- [ ] Made it possible to drag & drop charts in Dashboard layout to change their positions
- [ ] Add single URLs layout option

### How to test
1. Go to [deployment preview](https://vercel.live/open-feedback/visualization-tool-git-feat-add-layout-step-ixt1.vercel.app?via=pr-comment-visit-preview-link&passThrough=1).
2. Add `&flag__layoutStep=true` to the end of the browser URL and hit Enter.
3. Select a dataset and start making a chart.
4. See that you can publish the chart if you only made one chart.
5. Add another chart.
6. See that you have to proceed to layout options to publish your creation.
7. See that it's possible to add an overarching title and description to your creation.
8. See that it's possible to select from Tab or Dashboard layouts.
9. See that it's possible to select Vertical or Tall layout subtypes in the Dashboard layout.
10. Try to make more than 3 charts to see that Tall layout "recycles" its layout.
